### ISSUE TITLE
Add a11y-support check for nested ownership

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#1375]
```html
<div role="menu">
  <div><div role="menuitem">Item 1</div></div>
  <div role="presentation">
    <div role="menuitem">Item 1</div>
  </div>
</div>
```

Axe-core flags these correctly as failures. What it does not do however is report that these are failures because owning descendants isn't widely supported. We should add a check that can distinguish between "no owner/owned element at all" and "an owner/owned element, but done in a way that isn't accessibility supported".

I don't yet have a proposal on how to do this, but I'm sure we can come up with something. The newly introduced `aria.getOwner()` could help with that, and should probably get a few "unsupported" feature flags so that we can make this configurable.

**Comment by straker** [comment]
With the changes we made to [required-children](https://github.com/dequelabs/axe-core/pull/2394), this code no longer throws an error.

**Comment by straker** [comment]
Need to do some testing to verify if the `<div>` wrapping the first `menuitem` is a problem for screen readers.
